services/identity/service.go

```diff
@@ -143,35 +161,45 @@ func (s *Service) ProposeFetch(si *network.ServerIdentity, cnc *ProposeFetch) (n
 // that the voter is in the latest config.
 // An empty signature signifies that the vote has been rejected.
 func (s *Service) ProposeVote(si *network.ServerIdentity, v *ProposeVote) (network.Body, error) {
+	log.Lvl2(s, "Voting on proposal")
 	// First verify if the signature is legitimate
 	sid := s.getIdentityStorage(v.ID)
 	if sid == nil {
 		return nil, errors.New("Didn't find identity")
 	}
-	sid.Lock()
-	log.Lvl3("Voting on", sid.Proposed.Device)
-	owner, ok := sid.Latest.Device[v.Signer]
-	if !ok {
-		return nil, errors.New("Didn't find signer")
-	}
-	if sid.Proposed == nil {
-		return nil, errors.New("No proposed block")
-	}
-	hash, err := sid.Proposed.Hash()
-	if err != nil {
-		return nil, errors.New("Couldn't get hash")
-	}
-	if _, exists := sid.Votes[v.Signer]; exists {
-		return nil, errors.New("Already voted for that block")
-	}
-	log.Lvl3(v.Signer, "voted", v.Signature)
-	if v.Signature != nil {
-		err = crypto.VerifySchnorr(network.Suite, owner.Point, hash, *v.Signature)
+
+	// Putting this in a function because of the lock which needs to be held
+	// over all calls that might return an error.
```

Comment:
Wouldn't a simple `defer sid.Unlock`do the same job? Why does the `sid` needs to be locked in this scope but some lines later where it is also read but not modified (like here: https://github.com/dedis/cothority/pull/529/files#diff-2433496499fd4c0770b52cb82bb9d4caL181) the Lock isn't used.